rtmp_to_adaptive_hls/lib/rtmp_to_adaptive_hls/pipeline.ex

```diff
@@ -0,0 +1,100 @@
+defmodule Membrane.Demo.RtmpToAdaptiveHls do
+  use Membrane.Pipeline
+
+  alias Membrane.RTMP.SourceBin
+
+  @segment_duration Membrane.Time.seconds(4)
+  @transcode_targets [{720, 60}, {720, 30}, {360, 30}, {180, 30}, {180, 2}]
+
+  @impl true
+  def handle_init(_context, %{client_ref: client_ref}) do
+    structure = [
+      child(:src, %SourceBin{
+        client_ref: client_ref
+      }),
+      child(:sink, %Membrane.HTTPAdaptiveStream.SinkBin{
+        manifest_module: Membrane.HTTPAdaptiveStream.HLS,
+        target_window_duration: :infinity,
+        persist?: false,
+        storage: %Membrane.HTTPAdaptiveStream.Storages.FileStorage{directory: "output"}
+      }),
+      get_child(:src)
+      |> via_out(:audio)
+      |> via_in(Pad.ref(:input, "audio_master"),
+        options: [
+          encoding: :AAC,
+          segment_duration: @segment_duration
+        ]
+      )
+      |> get_child(:sink),
+      get_child(:src)
+      |> via_out(:video)
+      |> child(:tee_video, Membrane.Tee.Parallel)
+    ]
+
+    {[spec: structure], %{client_ref: client_ref}}
+  end
+
+  @impl true
+  def handle_info(
+        {:metadata_message, %Membrane.RTMP.Messages.SetDataFrame{} = message},
+        _ctx,
+        state
+      ) do
```

Comment:
I think it would be good to make sure that this callback doesn't get called more than once, which could happen if `SetDataFrame` is sent multiple times (I don't think RTMP disallows that)

Reply:
This is handled now by storing the metadata message in the client handler's state and overloading handle_info/3 to only get called if the metadata message has yet to arrive.